stdlib/mexpr/eq.mc

```diff
@@ -362,14 +362,44 @@ lang VarPatEq = VarPat
     else None ()
 end
 
-lang SeqTotPatEq
+lang SeqTotPatEq = SeqTotPat
   sem eqPat (env : Env) (free : Env) (patEnv : NameEnv) (lhs : Pat) =
-  -- TODO(dlunde,2020-09-29)
+  | PSeqTot {pats = ps2} ->
+    match lhs with PSeqTot {pats = ps1} then
+      if eqi (length ps2) (length ps1) then
+        let z = zipWith (lam p1. lam p2. (p1,p2)) ps1 ps2 in
+        optionFoldlM (lam fpEnv. lam ps.
+            match fpEnv with (f,p) then
+              eqPat env f p ps.0 ps.1 else None ())
+               (free,patEnv) z
+      else None ()
+    else None ()
 end
 
-lang SeqEdgPatEq
+lang SeqEdgPatEq = SeqEdgePat
+  -- NOTE(gcaylak,2020-10-02): 'SeqEdgPatEq' is called 'SeqEdgePat' in 'ast.mc'.
+  -- Should we change 'SeqEdgPatEq' to 'SeqEdgePatEq' as well?
```

Comment:
I originally named the pattern "edg" to match the length of "tot", but later regretted it as a poor abbreviation, and so changed to "edge"

Reply:
I believe this is my error (I copied the names from `ast.mc` before Viktor's changes). Could you fix this Gizem?

Reply:
Yes, I will fix that.